paasta_tools/cli/cmds/mark_for_deployment.py

```diff
@@ -527,7 +537,14 @@ def wait_for_deployment(service, deploy_group, git_sha, soa_dir, timeout):
 
     total_instances = 0
     clusters_data = []
+    api_endpoints = load_system_paasta_config().get_api_endpoints()
```

Comment:
In ~10 lines from this one there is a function which creates a number of threads to send async queries to clusters.

If I raise the exception in a child thread, it will only terminate that thread, but not the main process.
Therefore I will need to send a signal to other threads to terminate them.
And while such signal is being sent and processed all un-terminated threads still continue to print their output to stdout.

I'm certain that calling load_system_paasta_config().get_api_endpoints() here is the best way to fail fast, as PAASTA-10702 implies.